usecases/scenario_iterations_usecase.go

```diff
@@ -252,3 +318,34 @@ func replaceTriggerOrRule(scenarioAndIteration scenarios.ScenarioAndIteration,
 
 	return scenarioAndIteration, nil
 }
+
+func (usecase *ScenarioIterationUsecase) getScenarioVersion(
+	ctx context.Context,
+	exec repositories.Executor,
+	organizationId, scenarioId, iterationId string,
+) (int, error) {
+	scenarioIterations, err := usecase.repository.ListScenarioIterations(
+		ctx,
+		exec,
+		organizationId,
+		models.GetScenarioIterationFilters{ScenarioId: &scenarioId})
+	if err != nil {
+		return 0, err
+	}
+
+	for _, scenarioIteration := range scenarioIterations {
+		if scenarioIteration.Id == iterationId && scenarioIteration.Version != nil {
+			return *scenarioIteration.Version, nil
+		}
+	}
```

Comment:
Not required, or should return an error (not the current version).

It's due to the copy paste: the old method was doing two things in one, either retrieve the existing version number (= reactivate) or compute the next version to commit the draft (=activate)

Reply:
ah yes, my bad